ui/views.py

```diff
@@ -137,7 +137,10 @@ def upload(request, repo_slug):
     """
     Upload a OLX archive.
     """
-    repo = get_repo(repo_slug, request.user.id)
+    try:
+        repo = get_repo(repo_slug, request.user.id)
```

Comment:
Everything looks good except these additions -- the decorator means it can't happen.